dll/ntdll/ldr/ldrinit.c

```diff
@@ -2135,13 +2136,58 @@ LdrpInitializeProcess(IN PCONTEXT Context,
         if (FreeCurDir) RtlFreeUnicodeString(&CurrentDirectory);
     }
 
-    /* Check if we should look for a .local file 
-        FIXME: Thomas suggested that this check might actually be reversed, we should check this file
-               if the flag is NOT set. */
-    if (ProcessParameters && (ProcessParameters->Flags & RTL_USER_PROCESS_PARAMETERS_LOCAL_DLL_PATH))
+    /* Check if we should look for a .local file */
+    if (ProcessParameters && !(ProcessParameters->Flags & RTL_USER_PROCESS_PARAMETERS_LOCAL_DLL_PATH))
     {
-        /* FIXME */
-        DPRINT1("We don't support .local overrides yet\n");
+        UNICODE_STRING ImagePathName = Peb->ProcessParameters->ImagePathName;
+        WCHAR LocalBuffer[MAX_PATH];
+        UNICODE_STRING DotLocal;
+
+        ULONG RequiredSize;
+        RequiredSize = ImagePathName.Length + LdrpDotLocal.Length + sizeof(WCHAR);
+        if (RequiredSize <= sizeof(LocalBuffer))
+        {
+            RtlInitEmptyUnicodeString(&DotLocal, LocalBuffer, sizeof(LocalBuffer));
+        }
+        else if (RequiredSize <= UNICODE_STRING_MAX_BYTES)
+        {
+            DotLocal.Buffer = RtlAllocateHeap(Peb->ProcessHeap, 0, RequiredSize);
+            DotLocal.Length = 0;
+            DotLocal.MaximumLength = RequiredSize;
+            if (!DotLocal.Buffer)
+                DPRINT1("LDR: Failed to allocate memory for .local check\n");
+        }
+        else
+        {
+            DotLocal.Buffer = NULL;
+            DotLocal.Length = 0;
+            DotLocal.MaximumLength = 0;
+            DPRINT1("LDR: String too big for .local check\n");
+        }
+
+        if (DotLocal.Buffer)
```

Comment:
This code path isn't so big that it's a big deal, but it might read nicer with some early exits -- either with a goto label or by moving this into its own function.

Reply:
I'll move this entire block into a function.